### PR TITLE
Remove unused/unfinished classes

### PR DESCRIPTION
**Pull request by tsherif**
Remove `Sampler`, `Texture2DArray`, `pickModels`, `UniformBufferLayout`.

Moved code, docs and specs into 'wip' directory.

### CODE REVIEW

**Comment by ibgreen** [comment]
The most important parts of this PR is to explain what is happening in `Upgrade Guide`, please add that.

Also make sure you have run all examples and that they still work 

`UniformBufferLayout` - I'd really like to keep the `UniformBufferLayout` for now, as I see that as a big part of the near-term evolution of the framework as discussed yesterday. It does have test cases and in that sense is not an untested piece of code, so I'd like to have a thorough discussion of why that class does not have a place in the next release before cutting it, and I prefer we take that discussion after 7.0 release.

`pickModels` - I've gone back and forth about this, but Im still not happy about `pickModels`. `pickModels` was already commented out in the examples which leaves us with another mess (having commented non-functional code in examples). We could cut that code now in all the examples but then we have to add it all back in the next release, and that is a level of churn that I do not appreciate. I am really tempted to just restore pickModels the way it worked in 6.4 (a five line change) and leave it in for now, and then fix it properly by porting some of luma.gl picking in next release.

Finally there is some is additional WIP code inside e.g. `core` and `webgl` modules, suggest we also move that to the new top-level `wip` folder in this PR.

**Comment by tsherif** [comment]
> I'd really like to keep the UniformBufferLayout for now

Do you think we could update one of the examples to use it? That would be enough to make me more comfortable with its inclusion.

> was already commented out in the examples which leaves us with another mess 
(having commented non-functional code in examples)

My preference here would be for re-implementing picking in the examples manually, by just reading the pixels, then maintaining the previous code that wasn't really working.

> additional WIP code inside e.g. core and webgl modules

Which ones were you thinking?

**Comment by ibgreen** [comment]
> Which ones were you thinking?

Found only this one https://github.com/uber/luma.gl/tree/master/modules/core/wip

> My preference here would be for re-implementing picking in the examples manually

Sounds good!

> Do you think we could update one of the examples to use it

Great idea, maybe the transform feedback example as it is already webgl2?

File: examples/core/instancing/app.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ A luma.gl <code>Cube</code>, rendering 65,536 instances in a
 single GPU draw call using instanced vertex attributes.
 `;
 
+function getDevicePixelRatio() {
```
Comment (ibgreen) [inline examples/core/instancing/app.js]: I think the animationLoop provides this as a param

Comment (tsherif) [reply examples/core/instancing/app.js]: It provides the true/false param, but not the device pixel ratio itself: https://github.com/uber/luma.gl/blob/master/modules/core/src/core/animation-loop.js#L286-L308

(Maybe it should?)

Comment (ibgreen) [reply examples/core/instancing/app.js]: Yes, I think it should, in fact the ratio can very per frame (e.g. when zooming into a web page) so why not send it in every frame?

**Pull request merged**